node/network/pov-distribution/src/lib.rs

```diff
@@ -289,7 +366,21 @@ async fn handle_fetch(
 				return Ok(());
 			}
 			Entry::Vacant(e) => {
-				e.insert(vec![response_sender]);
+				if let Some(relevant_validators) = determine_relevant_validators(
+					ctx,
+					relay_parent,
+					descriptor.para_id,
+				).await? {
+					let new_connection_request = validator_discovery::connect_to_validators(
```

Comment:
Could it be too heavy to constantly call `connect_to_validators`? Does this call typically return instantly if we're already connected to those validators?

Reply:
I think our expectation is that we'll call this function N times per relay-parent, where N is the number of other validators in our group.

Reply:
connect_to_validators returns a `request` which is a stream that can be polled to retrieve the ids as we discover them (and instantly if already known), so it shouldn't be too heavy latency-wise unless we have too many revoked requests to process.

Reply:
OK, that's workable then!